File: duolingo_rum/Models/LearningSession.cs
using System;
using System.Collections.Generic;

namespace duolingo_rum.Models;

public partial class LearningSession
{
    public Guid Id { get; set; }

    public Guid UserId { get; set; }

    public int LanguageId { get; set; }

    public DateTime? StartedAt { get; set; }

    public DateTime? FinishedAt { get; set; }

    public int? DurationSec { get; set; }

    public int? WordsStudied { get; set; }

    public int? CorrectAnswers { get; set; }

    public int? WrongAnswers { get; set; }

    public int? XpEarned { get; set; }

    public virtual ICollection<ExerciseResult> ExerciseResults { get; set; } = new List<ExerciseResult>();

    public virtual Language Language { get; set; } = null!;

    public virtual User User { get; set; } = null!;
}

File: android-app/app/src/main/java/com/linguaai/app/models/LearningSession.kt
package com.linguaai.app.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class LearningSession(
    val id: String = "",
    @SerialName("user_id")
    val userId: String = "",
    @SerialName("language_id")
    val languageId: Int = 0,
    @SerialName("started_at")
    val startedAt: String? = null,
    @SerialName("finished_at")
    val finishedAt: String? = null,
    @SerialName("duration_sec")
    val durationSec: Int? = null,
    @SerialName("words_studied")
    val wordsStudied: Int? = 0,
    @SerialName("correct_answers")
    val correctAnswers: Int? = 0,
    @SerialName("wrong_answers")
    val wrongAnswers: Int? = 0,
    @SerialName("xp_earned")
    val xpEarned: Int? = 0
)
